lib/std/private/digitsutils.nim

```diff
@@ -28,13 +28,13 @@ const
 #       res.add $i
 #   doAssert res == digits100
 
-proc utoa2Digits*(buf: var openArray[char]; pos: int; digits: uint32) {.inline.} =
+proc utoa2Digits*[T: Ordinal](buf: var openArray[char]; pos: T; digits: uint32) {.inline.} =
```

Comment:
Why is this change required? `pos` should be fine as `int` even if `int` is mapped to 16 bits.

Reply:
I just tried https://github.com/nim-lang/Nim/commit/7456d09809dbd1b5752a0854798732d54b4763dc, in the end it still relevant to `FloatingDecimal32` which has `exponent: int32`, so to fix this will lose the possibility of string representation number larger than system bit width

Reply:
We don't need a string representation that supports more than the available address space.